### ISSUE TITLE
False positive SC2124 on `quoted_args="${@@Q}"`

### ISSUE THREAD

**Issue by l0b0** [koalaman/shellcheck#2778]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2124
- My shellcheck version (`shellcheck --version` or "online"): 0.9.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
aws-azure-login() {
    local quoted_args="${@@Q}"
    nix-shell --packages nodePackages.aws-azure-login --run "aws-azure-login $quoted_args"
}
```

#### Here's what shellcheck currently says:

```
In .bashrc line 155:
    local quoted_args="${@@Q}"
                      ^------^ SC2124 (warning): Assigning an array to a string! Assign as array, or use * instead of @ to concatenate.
```

#### Here's what I wanted or expected to see:

No warning.
